File: labellab_mobile/lib/model/classification.dart
import 'package:labellab_mobile/model/classification_label.dart';

class Classification {
  String id;
  String imageName;
  String imageUrl;
  List<ClassificationLabel> labels;
  DateTime createdAt;

  Classification();

  Classification.fromJson(dynamic json, {String staticEndpoint}) {
    id = json["id"].toString();
    imageName = json["image_name"];
    imageUrl = (staticEndpoint != null ? staticEndpoint : "") +
        json["user_id"].toString() +
        "/" +
        json["image_url"];
    createdAt = DateTime.parse(json["classified_at"]);
    labels = [
      ClassificationLabel.fromValues(
          name: json["label"], confidence: json["confidence"])
    ].toList();
  }
}

File: labellab-flask/api/serializers/classification.py
from marshmallow import Schema, fields

from api.extensions import db, ma


class ClassificationSchema(ma.ModelSchema):
    # Serializer for Classification

    id = fields.Int(dump_only=True)
    image_name = fields.Str()
    image_url = fields.Str()
    label = fields.Str()
    confidence = fields.Float()
    user_id = fields.Int(dump_only=True)
    classified_at = fields.DateTime()
